cpp/tests/io/csv_test.cpp

```diff
@@ -2194,4 +2194,38 @@ TEST_F(CsvReaderTest, DtypesMapInvalid)
   EXPECT_THROW(cudf_io::read_csv(in_opts), cudf::logic_error);
 }
 
+TEST_F(CsvReaderTest, CsvDefaultOptionsWriteReadMatch)
+{
+  auto filepath = temp_env->get_temp_dir() + "issue.csv";
+
+  // make up some kind of dataframe
+  auto constexpr num_rows = 10;
+
+  auto int32_column       = []() {
+    auto values = random_values<int32_t>(num_rows);
+    return column_wrapper<int32_t>(values.begin(), values.end());
+  }();
+  auto int64_column = []() {
+    auto values = random_values<int64_t>(num_rows);
+    return column_wrapper<int64_t>(values.begin(), values.end());
+  }();
+  std::vector<cudf::column_view> input_columns{int32_column, int64_column};
+
+  cudf::table_view original_table{input_columns};
+  cudf::table_view new_table;
```

Comment:
declaring a variable without initializing it is usually a code smell. see the comment on `csv_reader_options` below for where this smell originates...